lib/internal/main/test_runner.js

```diff
@@ -31,8 +31,13 @@ if (isUsingInspector() && options.isolation === 'process') {
 options.globPatterns = ArrayPrototypeSlice(process.argv, 1);
 
 debug('test runner configuration:', options);
-run(options).on('test:fail', (data) => {
+const ran = run(options);
+ran.on('test:fail', (data) => {
   if (data.todo === undefined || data.todo === false) {
     process.exitCode = kGenericUserError;
   }
 });
+
+ran.on('test:coverage', ({ summary: { thresholds: { line, branch, function: func } } }) => {
```

Comment:
I don't think we should do this. Trying to identify every possible way that the exit code can be set will be a major pain to users of this interface. I've opened https://github.com/nodejs/node/pull/54851 with a different approach.